cmd/kosli/attestSonar.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/kosli-dev/cli/internal/requests"
+	"github.com/kosli-dev/cli/internal/sonar"
+	"github.com/spf13/cobra"
+)
+
+type SonarAttestationPayload struct {
+	*CommonAttestationPayload
+	SonarResults *sonar.SonarResults `json:"sonar_results"`
+}
+
+// It is also possible to provide a branch or PR number, but not necessary - how do we handle this here?
+type attestSonarOptions struct {
+	*CommonAttestationOptions
+	projectKey string
+	apiToken   string
+	payload    SonarAttestationPayload
+}
+
+const attestSonarShortDesc = `Report a sonarcloud or sonarqube attestation to an artifact or a trail in a Kosli flow.  `
+
+const attestSonarLongDesc = attestSonarShortDesc + attestationBindingDesc
+
+const attestSonarExample = `
+# report a sonarcloud attestation about a trail:
+kosli attest sonar \
+	--name yourAttestationName \
+	--flow yourFlowName \
+	--trail yourTrailName \
+	--sonar-project-key yourSonarProjectKey \
+	--sonar-api-token yourSonarAPIToken \
+	--api-token yourAPIToken \
+	--org yourOrgName \
+
+# report a sonarqube attestation about a trail:
+kosli attest sonar \
+	--name yourAttestationName \
+	--flow yourFlowName \
+	--trail yourTrailName \
+	--sonar-project-key yourSonarProjectKey \
+	--sonar-api-token yourSonarAPIToken \
+	--sonar-server-url yourSonarQubeServerURL \
```

Comment:
is this example supported?

Reply:
Whoops, good catch. Forgot to get rid of that. Will do now.